reference/preconditioner/jacobi_kernels.cpp

```diff
@@ -724,6 +724,59 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_JACOBI_CONVERT_TO_DENSE_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void scalar_l1(std::shared_ptr<const DefaultExecutor> exec,
+               const matrix::Csr<ValueType, IndexType>* csr,
+               matrix::Diagonal<ValueType>* diag)
+{
+    for (IndexType row = 0; row < csr->get_size()[0]; row++) {
+        auto off_diag = zero<ValueType>();
+        for (auto i = csr->get_const_row_ptrs()[row];
+             i < csr->get_const_row_ptrs()[row + 1]; i++) {
+            if (csr->get_const_col_idxs()[i] == row) {
+                continue;
+            }
+            off_diag += abs(csr->get_const_values()[i]);
+        }
+        diag->get_values()[row] += off_diag;
```

Comment:
```suggestion
        diag->get_values()[row] = abs(diag->get_values()[row]) + off_diag;
```

Reply:
I think we discussed this a little bit before, but did not get a reasonable choice when the diagonal values are negative.
L1 is originally designed for postive diagonal value